app/components/score_component.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class ScoreComponent < ViewComponent::Base
+  NOT_SCORED = -2
+
+  def initialize(score:, show_decimal_score: false)
+    @score = score
+    @show_decimal_score = show_decimal_score
+  end
+
+  def render?
+    @score.present?
+  end
+
+  private
+
+  def show_decimal_score?
+    @show_decimal_score
+  end
+
+  def highlight_stars?
+    # We only highlight the stars when a decimal score is displayed
+    show_decimal_score?
+  end
+
+  def scored?
+    @score != NOT_SCORED
+  end
```

Comment:
I'm not sure why @score != -2 is the indicator that a category has been scored, rather than @score >= 0, for example. Possibly linked to the '-2' question above!

Reply:
Good shout - I'll change this 👍🏻